src/hooks/useWindowSize.jsx

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+function useWindowSize() {
+  // Initialize state with undefined width/height so server and client renders match
+  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
+  const [windowSize, setWindowSize] = useState({
+    width: undefined,
+    height: undefined,
+  });
+
+  useEffect(() => {
+    // Handler to call on window resize
+    function handleResize() {
+      // Set window width/height to state
+      setWindowSize({
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    }
+
+    // Add event listener
+    window.addEventListener('resize', handleResize);
```

Comment:
Avoid using `window` and instead use `global`. This change preserves compatibility with server-rendering or some pre-rendering done by Gatsby.